### PR TITLE
Addition and Addition with Assignment prop tests for felts

### PR DESCRIPTION
**Pull request by martinacantaro**
# Addition and Addition with Assignment prop tests for felts

## Description

This PR is part of https://github.com/lambdaclass/cairo-rs/issues/700: Add property based testing using proptest to ensure operations done with felts are working correctly. These are all the current operations available for felts.

This PR adds the following operations:
- Add
- AddAssign

## Checklist
- [x] Linked to Github Issue
- [x] Unit tests added
- [-] Integration tests added.
- [x] This change requires new documentation.
  - [x] Documentation has been added/updated. The tests are documented in the comments

### CODE REVIEW

**Review by Oppen — APPROVED** [review]

File: felt/src/lib.rs [hunk 1]
```diff
@@ -167,6 +167,30 @@ mod test {
...
+        #[test]
+        // Property-based test that ensures, for 100 {x} and {y} values that are randomly generated each time tests are run, that an addition between two felts {x} and {y} and doesn't fall outside the range [0, p]. The values of {x} and {y} can be either [0] or a very large number.
+        fn add_in_range(ref x in "(0|[1-9][0-9]*)", ref y in "(0|[1-9][0-9]*)") {
+            let x = &Felt::parse_bytes(x.as_bytes(), 10).unwrap();
+            let y = &Felt::parse_bytes(y.as_bytes(), 10).unwrap();
+            let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
+
+            let result = x + y;
+            let as_uint = &result.to_biguint();
+            println!("x: {}, y: {}, result: {}", x, y, result);
```
Comment (Juan-M-V) [inline felt/src/lib.rs:179]: Can this print be removed?

Comment (martinacantaro) [reply felt/src/lib.rs:179]: definitely! that was a testing print

**Review by Oppen — APPROVED** [review]

File: felt/src/lib.rs (hunk 1 above)
Comment (martinacantaro) [reply felt/src/lib.rs]: Done!

**Comment by martinacantaro** [comment]
Ready to go 🫡

**Comment by Oppen** [comment]
Is this based on the right PR? I get tons of unrelated changes.

**Review by Juan-M-V — APPROVED** [review]

**Pull request merged**